tests/test_mc_comm.py

```diff
@@ -269,3 +434,201 @@ def test_McCommunicationProcess_register_magic_word__raises_error_if_search_exce
     mc_process.set_board_connection(board_idx, simulator)
     with pytest.raises(SerialCommPacketRegistrationSearchExhaustedError):
         invoke_process_run_and_check_errors(mc_process)
+
+
+def test_McCommunicationProcess_register_magic_word__does_not_try_to_register_when_not_connected_to_anything(
+    four_board_mc_comm_process,
+):
+    mc_process = four_board_mc_comm_process["mc_process"]
+    invoke_process_run_and_check_errors(mc_process)
+
+    assert mc_process.is_registered_with_serial_comm(0) is False
+    assert mc_process.is_registered_with_serial_comm(1) is False
+    assert mc_process.is_registered_with_serial_comm(2) is False
+    assert mc_process.is_registered_with_serial_comm(3) is False
+
+
+def test_McCommunicationProcess__raises_error_if_magic_word_is_incorrect_in_packet_after_previous_magic_word_has_been_registered(
+    four_board_mc_comm_process, mantarray_mc_simulator_no_beacon, mocker
+):
+    mocker.patch(
```

Comment:
note for future--could probably extract this as a common fixture `patched_print`